src/helpers.rs

```diff
@@ -905,24 +905,26 @@ impl<'a, 'mir: 'a, 'tcx: 'a + 'mir> CurrentSpan<'a, 'mir, 'tcx> {
     /// This function is backed by a cache, and can be assumed to be very fast.
     pub fn get(&mut self) -> Span {
         let idx = self.current_frame_idx();
-        Self::frame_span(self.machine, idx)
+        self.stack().get(idx).map(Frame::current_span).unwrap_or(rustc_span::DUMMY_SP)
     }
 
-    /// Similar to `CurrentSpan::get`, but retrieves the parent frame of the first non-local frame.
+    /// Similar to `CurrentSpan::get`, but if the last frame is local, we return its parent.
```

Comment:
Not "last" frame but "current" frame, right?

Reply:
Ah no you mean "last" as in "top". I can see how that matches the implementation. I don't quite understand why that is always what we want for these diagnostics though.

Reply:
Maybe I should say "top". I and others have often found it confusing what order the stacks in the interpreter are in (I'm sure it's totally sensible in the implementation), and I can never remember if stacks normally grow up or down on modern computers.

We're looking for the function call that we're entering here, because while the problem may arise inside the call, the context that the programmer needs is nearly always the location of the call that prompted the error. So if we only provide the span inside the function, you'll still need to re-run the program and track the tag or the call in order to figure out how to change the code to make it valid.

Reply:
Okay so we are looking for the caller *of the current operation* (and then we walk down the stack to find the topmost local frame), but previously this would return the caller of the topmost local frame, which made no sense.